src/rgw/rgw_rest_s3.cc

```diff
@@ -1809,43 +1809,65 @@ void RGWListBucket_ObjStore_S3::send_response()
     s->formatter->dump_string("EncodingType", "url");
     encode_key = true;
   }
+  
   RGWListBucket_ObjStore_S3::send_common_response();
-    if (op_ret >= 0) {
-      vector<rgw_bucket_dir_entry>::iterator iter;
-      for (iter = objs.begin(); iter != objs.end(); ++iter) {
-        rgw_obj_key key(iter->key);
-        s->formatter->open_array_section("Contents");
-        if (encode_key) {
-          string key_name;
-          url_encode(key.name, key_name);
-          s->formatter->dump_string("Key", key_name);
+  
+  if (op_ret >= 0) {
+    if (s->format == RGWFormat::JSON) {
+      s->formatter->open_array_section("Contents");
+    }
+    vector<rgw_bucket_dir_entry>::iterator iter;
+    for (iter = objs.begin(); iter != objs.end(); ++iter) {
+
+      rgw_obj_key key(iter->key);
+      std::string key_name;
+
+      if (encode_key) {
+	url_encode(key.name, key_name);
       } else {
-          s->formatter->dump_string("Key", key.name);
+	key_name = key.name;
       }
-        dump_time(s, "LastModified", iter->meta.mtime);
-        s->formatter->dump_format("ETag", "\"%s\"", iter->meta.etag.c_str());
-        s->formatter->dump_int("Size", iter->meta.accounted_size);
-        auto& storage_class = rgw_placement_rule::get_canonical_storage_class(iter->meta.storage_class);
-        s->formatter->dump_string("StorageClass", storage_class.c_str());
-        dump_owner(s, rgw_user(iter->meta.owner), iter->meta.owner_display_name);
-        if (s->system_request) {
-          s->formatter->dump_string("RgwxTag", iter->tag);
-      }
-        if (iter->meta.appendable) {
-          s->formatter->dump_string("Type", "Appendable");
+      /* conditionally format JSON in the obvious way--I'm unsure if
+       * AWS actually does this */
+      if (s->format == RGWFormat::XML) {
+	s->formatter->open_array_section("Contents");
+	s->formatter->dump_string("Key", key_name);
       } else {
-          s->formatter->dump_string("Type", "Normal");
+	// json
+	s->formatter->open_object_section("dummy");
+	s->formatter->open_object_section(key_name);
```

Comment:
hmm, so the json response will have the outer "Contents" array of unnamed "dummy" objects, and each of those objects contain a single key/value like this?
```json
"Contents": [
  {
    "keyname1": {
      "LastModified": "timestamp",
      "ETag": "etag",
    },
  },
  {
    "keyname2": {
      "LastModified": "timestamp",
      "ETag": "etag",
    },
  },
]
```
i would have expected "Key" to stay at the same level as the rest of the per-object fields like this:
```json
"Contents": [
  {
    "Key": "keyname1",
    "LastModified": "timestamp",
    "ETag": "etag",
  },
  {
    "Key": "keyname2",
    "LastModified": "timestamp",
    "ETag": "etag",
  },
```
that seems like the more direct translation from the xml format

i played around with awscli, and found that it outputs its results in json format by default (although the client is just converting the xml response into json), and this is how it looks:
> $ aws --endpoint-url http://localhost:8000 s3api list-objects --bucket testbucket
```json
{
    "Contents": [
        {
            "Key": "ceph.conf",
            "LastModified": "2022-08-16T16:22:08.109Z",
            "ETag": "\"94697f9200da0f4f5ce3d0959f822275\"",
            "Size": 5371,
            "StorageClass": "STANDARD",
            "Owner": {
                "DisplayName": "M. Tester",
                "ID": "testid"
            }
        },
        {
            "Key": "keyring",
            "LastModified": "2022-08-16T16:21:54.894Z",
            "ETag": "\"7bfa32696bf5a5df785ce3039aefc2ea\"",
            "Size": 294,
            "StorageClass": "STANDARD",
            "Owner": {
                "DisplayName": "M. Tester",
                "ID": "testid"
            }
        }
    ]
}
```

Reply:
ok, thanks for that--it should definitely match what awscli thinks